src/pages/List.js

```diff
@@ -1,20 +1,32 @@
 import { db } from '../lib/firebase';
 import { useCollection } from 'react-firebase-hooks/firestore';
 import { useHistory } from 'react-router-dom';
+import calculateEstimate from '../lib/estimates';
 
 export default function List({ token }) {
   const history = useHistory();
   const [listItem, loading, error] = useCollection(db.collection(token), {
     snapshotListenOptions: { includeMetadataChanges: true },
   });
 
-  const markItemPurchased = (e, id) => {
+  const markItemPurchased = (e, id, itemData) => {
     const elapsedMilliseconds = Date.now();
+    const latestInterval = elapsedMilliseconds - itemData.last_purchased;
 
     if (e.target.checked === true) {
-      db.collection(token).doc(id).update({
-        last_purchased: elapsedMilliseconds,
-      });
+      const lastEstimate = calculateEstimate(
+        itemData.last_estimate,
+        latestInterval,
+        itemData.times_purchased,
+      );
```

Comment:
`calculateEstimate()` requires:
> * @param {Number} lastEstimate The last estimated purchase interval
> * @param {Number} latestInterval The number of days between the most recent and previous purchases
> * @param {Number} numberOfPurchases Total number of purchases for the item
> * @return {Number} Estimated number of days until the next purchase

Things to consider:
- Does line 14 give me number of days? 
- What would the lastEstimate value be if this is the first time you mark the item purchased? Would it make sense to use `purcase_frequency` to calculate your first estimate?

Reply:
I'm seeing this behaviour too. I think maybe I assumed the helper function `calcuateEstimate` was handling this, based on how it checks for `if (numberOfPurchases >= 2)` and `returns latestInterval`.

Line 14 currently gives you the difference in milliseconds between how many milliseconds have passed since Januaray, 1st 1970 and when the item was last purchased (`itemData.last_purchased`).

So I can see that the first time an item is purchased, `itemData.last_purchased` will be `null`  and that will give the time ellapsed since January, 1st 1970. Obviously not what we want!

So as far as using `purchase_frequency` to calculate our first estimate, I'm not sure! Is that the functionality we are looking for. We would either be checking against 7, 14 or 30 days, which will generate an average based on that the first time. Does that seem right? I'll need guidance/direction on that.

The other thought I had is possibly initializing the Item in the database with a `created_at` timestamp that we can check against. We could use the `created_at` property when `times_purchased === 0`. 

But I am interested to know if using `purchase_frequency` makes more sense, in terms of desired functionality of the app.

Reply:
I see what you're saying. It's not a bad idea. And just so we're clear, we would ONLY be doing this when `times_purchased` === `0`, correct?

Reply:
Yup!

Reply:
I have updated `markItemPurchased`  with `initialInterval` to calculate the user's `purchase_frequency` selection of either 7, 14 or 30 days in milleconds. Then `if (itemData.times_purchased === 0)` update the item with `initialEstimate` using `initialInterval`.

I think I'm close, but I'm not sure this is exactly what we want. The `last_estimate` property in the database will simply be the `purchase_frequency` property multiplied by `86400000`.   Each subsequent time the item is marked purchased, the `calculateEstimate` helper function will use the `last_estimate` property to calcuate it's next purchase estimate.

But I'm just having a hard time wrapping my brain around is that the intended functionality.

```javascript
const markItemPurchased = (e, id, itemData) => {
    const currentTimestamp = Date.now();
    const initialInterval = itemData.purchase_frequency * 86400000;
    const latestInterval = currentTimestamp - itemData.last_purchased;

    if (e.target.checked === true) {
      if (itemData.times_purchased === 0) {
        const initialEstimate = calculateEstimate(
          itemData.last_estimate,
          initialInterval,
          itemData.times_purchased,
        );
        db.collection(token)
          .doc(id)
          .update({
            last_purchased: currentTimestamp,
            times_purchased: itemData.times_purchased + 1,
            last_estimate: initialEstimate,
          });
      } else {
        const latestEstimate = calculateEstimate(
          itemData.last_estimate,
          latestInterval,
          itemData.times_purchased,
        );
        db.collection(token)
          .doc(id)
          .update({
            last_purchased: currentTimestamp,
            times_purchased: itemData.times_purchased + 1,
            last_estimate: latestEstimate,
          });
      }
    }
  };
```